File: openqa-dashboard-ui/src/types/execution.ts
export interface Execution {
  id: string;
  name: string;
  buildNumber?: string;
  branch?: string;
  platform?: string;
  environment?: string;
  executionType?: string;
  status: string;
  startTime?: string;
  endTime?: string;
  durationMs: number;
  passCount: number;
  failCount: number;
  skipCount: number;
  totalCount: number;
  reportPath?: string;
  triggerSource?: string;
  mavenCommand?: string;
  features?: FeatureSummary[];
  scenarios?: ScenarioInfo[];
  createdAt: string;
}

export interface FeatureSummary {
  id: string;
  featureName: string;
  status: string;
  durationMs: number;
  passCount: number;
  failCount: number;
  skipCount: number;
}

export interface ScenarioInfo {
  id: string;
  featureId?: string;
  scenarioName: string;
  tags?: string;
  status: string;
  durationMs: number;
  failureReason?: string;
  deviceName?: string;
}

export interface ExecutionSummary {
  id: string;
  name: string;
  status: string;
  executionType: string;
  platform: string;
  startTime: string;
  endTime: string;
  durationMs: number;
  passCount: number;
  failCount: number;
  skipCount: number;
  totalCount: number;
  triggerSource: string;
  createdAt: string;
}

export interface Device {
  id: string;
  deviceId: string;
  brand: string;
  model: string;
  osVersion: string;
  platform: string;
  status: string;
  lastSeen: string;
}

export interface ReportItem {
  id: string;
  executionId: string;
  name: string;
  filePath: string;
  fileSize: number;
  createdAt: string;
}

File: openqa-dashboard-core/src/main/java/com/openqa/dashboard/model/entity/Execution.java
package com.openqa.dashboard.model.entity;

import jakarta.persistence.*;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "executions")
public class Execution {

    @Id
    private String id;

    @Column(name = "name")
    private String name;

    @Column(name = "framework")
    private String framework = "CUCUMBER";

    @Column(name = "branch")
    private String branch;

    @Column(name = "machine")
    private String machine;

    @Column(name = "platform")
    private String platform;

    @Column(name = "environment")
    private String environment;

    @Column(name = "build_number")
    private String buildNumber;

    @Column(name = "status")
    private String status = "PENDING";

    @Column(name = "execution_type")
    private String executionType = "REGRESSION";

    @Column(name = "start_time")
    private String startTime;

    @Column(name = "end_time")
    private String endTime;

    @Column(name = "duration_ms")
    private Long durationMs = 0L;

    @Column(name = "pass_count")
    private Integer passCount = 0;

    @Column(name = "fail_count")
    private Integer failCount = 0;

    @Column(name = "skip_count")
    private Integer skipCount = 0;

    @Column(name = "total_count")
    private Integer totalCount = 0;

    @Column(name = "report_path")
    private String reportPath;

    @Column(name = "log_path")
    private String logPath;

    @Column(name = "trigger_source")
    private String triggerSource = "MANUAL";

    @Column(name = "maven_command")
    private String mavenCommand;

    @Column(name = "created_at")
    private String createdAt;

    @Column(name = "updated_at")
    private String updatedAt;

    public Execution() {
        this.id = UUID.randomUUID().toString();
        this.createdAt = Instant.now().toString();
        this.updatedAt = this.createdAt;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = Instant.now().toString();
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getFramework() { return framework; }
    public void setFramework(String framework) { this.framework = framework; }
    public String getBranch() { return branch; }
    public void setBranch(String branch) { this.branch = branch; }
    public String getMachine() { return machine; }
    public void setMachine(String machine) { this.machine = machine; }
    public String getPlatform() { return platform; }
    public void setPlatform(String platform) { this.platform = platform; }
    public String getEnvironment() { return environment; }
    public void setEnvironment(String environment) { this.environment = environment; }
    public String getBuildNumber() { return buildNumber; }
    public void setBuildNumber(String buildNumber) { this.buildNumber = buildNumber; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getExecutionType() { return executionType; }
    public void setExecutionType(String executionType) { this.executionType = executionType; }
    public String getStartTime() { return startTime; }
    public void setStartTime(String startTime) { this.startTime = startTime; }
    public String getEndTime() { return endTime; }
    public void setEndTime(String endTime) { this.endTime = endTime; }
    public Long getDurationMs() { return durationMs; }
    public void setDurationMs(Long durationMs) { this.durationMs = durationMs; }
    public Integer getPassCount() { return passCount; }
    public void setPassCount(Integer passCount) { this.passCount = passCount; }
    public Integer getFailCount() { return failCount; }
    public void setFailCount(Integer failCount) { this.failCount = failCount; }
    public Integer getSkipCount() { return skipCount; }
    public void setSkipCount(Integer skipCount) { this.skipCount = skipCount; }
    public Integer getTotalCount() { return totalCount; }
    public void setTotalCount(Integer totalCount) { this.totalCount = totalCount; }
    public String getReportPath() { return reportPath; }
    public void setReportPath(String reportPath) { this.reportPath = reportPath; }
    public String getLogPath() { return logPath; }
    public void setLogPath(String logPath) { this.logPath = logPath; }
    public String getTriggerSource() { return triggerSource; }
    public void setTriggerSource(String triggerSource) { this.triggerSource = triggerSource; }
    public String getMavenCommand() { return mavenCommand; }
    public void setMavenCommand(String mavenCommand) { this.mavenCommand = mavenCommand; }
    public String getCreatedAt() { return createdAt; }
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }
    public String getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(String updatedAt) { this.updatedAt = updatedAt; }
}
